plugins/powerpc/powerpc_arith.ml

```diff
@@ -0,0 +1,420 @@
+
+open Powerpc.Std
+
+(** Fixed-Point Arithmetic Instructions - Negate
+    Page 72 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 00 d0   neg  r1, r2
+    7c 22 00 d1   neg. r1, r2 *)
+let neg cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  RTL.[ rt := lnot ra + one ]
+
+(** Fixed-Point Arithmetic Instructions - Substract From
+    Page 69 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 18 50    subf  r1, r2, r3
+    7c 22 18 51    subf. r1, r2, r3 *)
+let subf cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  let rb = unsigned cpu.reg ops.(2) in
+  RTL.[rt := (lnot ra) + rb + one]
```

Comment:
there is a strange controversy between addition and subtraction. In addition you're using signed operands while in subtraction they are unsigned. On the first glance, this looks incorrect. Also addition and subtraction are idempotent to the signedness of their operands. Probably it only matters when conditional flags are involved, especially when you express them via the comparison operation, that are indeed dependent on the interpretation of the sign bit. And my intuition here tells me, that we shall compare unsigned. Please, can you provide some insight on this?